src/lib/sap_ha/configuration/cluster.rb

```diff
@@ -278,6 +278,16 @@ def other_nodes_ext
         end.compact
       end
 
+      def get_primary_on_primary
+        SapHA::System::Network.ip_addresses.each do |my_ip|
+	  @nodes.map do |_, node|
+	    if node[:ip_ring1] == my_ip
+	      return node[:host_name]
+	    end
+	  end
+	end
```

Comment:
Be careful, if that `return` above is not called then the result of the function is the result of the top level `each` call. And that is the `SapHA::System::Network.ip_addresses` list itself.

To be safe I'd add `return nil` here. Or if that case should not really happen you might raise an exception here which will be caught by YaST top level exception handler and an error will be reported to user.

Reply:
BTW here is short summary about `return`, `break` and `next` in blocks: https://makandracards.com/makandra/46939-ruby-a-small-summary-of-what-return-break-and-next-means-for-blocks

Reply:
Thanks. Proposed change applied.